### PR TITLE
add go tests

### PR DESCRIPTION
**Pull request by dergeberl**
- move k8s client out of run to be able to test this function with a k8s fake client
- add tests for rename-pvc
- add github action for tests

### CODE REVIEW

File: pkg/renamepvc/renamepvc_test.go [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+	// the fake client did not set volumes to bound
+	// add reactor to set every volume to Phase bound
+	client.PrependReactor(
+		"create",
+		"persistentvolumeclaims",
+		func(action k8sTesting.Action) (handled bool, ret runtime.Object, err error) {
+			ret = action.(k8sTesting.CreateAction).GetObject()
+			pvc, ok := ret.(*corev1.PersistentVolumeClaim)
+			if !ok {
+				return
```
Comment (brumhard) [inline pkg/renamepvc/renamepvc_test.go:172]: no naked returns pls

File: pkg/renamepvc/renamepvc_test.go [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+
+// initTestSetup returns renamePVCOptions initialized with a kubernetes fake client and created oldPvc for test
+func initTestSetup(
+	ctx context.Context,
+	oldName, sourceNamespace string,
+	newName, targetNamespace string,
+	extraObjects ...runtime.Object,
+) (renamePVCOptions, error) {
+	// init client
+	streams, _, _, _ := genericclioptions.NewTestIOStreams() //nolint:dogsled // not needed for tests
```
Comment (brumhard) [inline pkg/renamepvc/renamepvc_test.go:161]: We can just add this to the overall linting config to exclude dogsled from test files

File: pkg/renamepvc/renamepvc_test.go [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+		}
+
+		err = o.run(ctx)
+		if !errors.Is(err, ErrVolumeMounted) {
+			t.Errorf("expect %q error - but got %q", ErrVolumeMounted, err)
+		}
+	})
+
+	t.Run("Test rename-pvc with running pod - fail", func(t *testing.T) {
+		ctx := context.Background()
```
Comment (brumhard) [inline pkg/renamepvc/renamepvc_test.go:93]: Ctx can be initialized once at the beginning

File: pkg/renamepvc/renamepvc_test.go [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+			t.Errorf("testsetup failed - got error %q", err)
+		}
+
+		err = o.run(ctx)
+		if !errors.Is(err, ErrVolumeMounted) {
+			t.Errorf("expect %q error - but got %q", ErrVolumeMounted, err)
+		}
+	})
+
+	t.Run("Test rename-pvc with running pod - fail", func(t *testing.T) {
```
Comment (brumhard) [inline pkg/renamepvc/renamepvc_test.go:92]: Test name is not correct.

File: pkg/renamepvc/renamepvc.go [hunk 5]
```diff
@@ -21,15 +21,16 @@ import (
 )
 
 var (
-	ErrNotBound             = errors.New("new PVC did not get bound")
-	ErrAcceptationNotYes    = errors.New("conformation was not successful please type in yes to continue")
-	ErrAcceptationUnknown   = errors.New("please type yes or no")
-	ErrVolumeAlreadyMounted = errors.New("volume already mounted")
+	ErrNotBound           = errors.New("new PVC did not get bound")
+	ErrAcceptationNotYes  = errors.New("conformation was not successful please type in yes to continue")
```
Comment (brumhard) [inline pkg/renamepvc/renamepvc.go:25]: I think that error should be renamed

**Review by brumhard — CHANGES_REQUESTED** [review]

File: pkg/renamepvc/renamepvc.go (hunk 5 above)
Comment (dergeberl) [reply pkg/renamepvc/renamepvc.go]: I am not sure what you mean exactly, I changed it and I also changed the naming from `Acceptation` to `Confirmation` to be align with the code.

Comment (brumhard) [reply pkg/renamepvc/renamepvc.go]: That's exactly it, nice one 👍🏻

**Pull request merged**